lib/client.js

```diff
@@ -1278,6 +1281,128 @@ function _resume (client, sync) {
   }
 }
 
+async function writeIterable (client, request, socket, contentLength, header, expectsPayload) {
+  const { body } = request
+  assert(contentLength !== 0 || client[kRunning] === 0, 'iterator body cannot be pipelined')
+  socket[kWriting] = true
+
+  let callback = noop
+  function onDrain (err) {
+    if (!error && err) {
+      error = err
+    }
+    const actualCallback = callback
+    callback = noop
+    actualCallback()
+  }
+
+  const waitForDrain = () => new Promise((resolve) => {
+    assert(callback === noop)
+    callback = resolve
+  })
+
+  let bytesWritten = 0
+  let error
+  try {
+    socket.on('error', onDrain)
+      .on('drain', onDrain)
+    for await (const chunk of body) {
+      if (error || socket.destroyed) {
+        return
+      }
+
+      const len = Buffer.byteLength(chunk)
+      if (!len) {
+        continue
+      }
+
+      // TODO: What if not ended and bytesWritten === contentLength?
+      // We should defer writing chunks.
+      if (contentLength !== null && bytesWritten + len > contentLength) {
+        if (client[kStrictContentLength]) {
+          throw new RequestContentLengthMismatchError()
+        }
+
+        process.emitWarning(new RequestContentLengthMismatchError())
+      }
+
+      if (bytesWritten === 0) {
+        if (!expectsPayload) {
+          socket[kReset] = true
+        }
+
+        if (contentLength === null) {
+          socket.write(`${header}transfer-encoding: chunked\r\n`, 'ascii')
+        } else {
+          socket.write(`${header}content-length: ${contentLength}\r\n\r\n`, 'ascii')
+        }
+      }
+
+      if (contentLength === null) {
+        socket.write(`\r\n${len.toString(16)}\r\n`, 'ascii')
+      }
+
+      bytesWritten += len
+
+      if (socket.write(chunk)) {
+        continue
+      } else if (socket.destroyed || error) {
+        return
+      } else {
+        await waitForDrain()
+        if (error) {
+          return
+        }
+      }
+    }
+  } catch (err) {
+    error = err
+  } finally {
+    socket[kWriting] = false
+
+    if (!error && contentLength !== null && bytesWritten !== contentLength) {
+      if (client[kStrictContentLength]) {
+        error = new RequestContentLengthMismatchError()
+      } else {
+        process.emitWarning(new RequestContentLengthMismatchError())
+      }
+    }
+
+    socket.off('error', onDrain)
+      .off('drain', onDrain)
+    if (error) {
+      assert(client[kRunning] <= 1, 'pipeline should only contain this request')
+      util.destroy(socket, error)
+    }
+
+    if (!socket.destroyed) {
```

Comment:
```suggestion
    if (socket.destroyed) {
      return
    }
```
reduce nesting

Reply:
Couldn't do it because it's in a finally, and the linter doesn't allow returns in a finally as it's unsafe. Is it OK to add a `eslint-disable-next-line`?